src/world_interface/motor/base_motor.cpp

```diff
@@ -0,0 +1,81 @@
+#include "base_motor.h"
+
+using namespace std::chrono_literals;
+
+namespace robot {
+base_motor::base_motor(robot::types::motorid_t motor, bool hasPosSensor)
+	: motor_id(motor), posSensor(hasPosSensor) {
+	// create scheduler if needed
+	std::lock_guard<std::mutex> lg(schedulerMutex);
+	if (!pSched) {
+		pSched.emplace();
+	}
+}
+
+bool base_motor::hasPositionSensor() const {
+	return posSensor;
+}
+
+void base_motor::setMotorVel(int32_t targetVel) {
+	// create a velocity controller if it's not already initialized
+	if (!velController) {
+		constructVelController();
+	}
+
+	// set velocity target
+	navtypes::Vectord<1> velocityVector{targetVel};
+	types::datatime_t currTime = types::dataclock::now();
+	velController->setTarget(currTime, velocityVector);
+
+	// check to see if the event exists. if yes, unschedule it
+	base_motor::unscheduleVelocityEvent();
+
+	// schedule position event
+	velEventID = pSched->scheduleEvent(100ms, [&]() -> void {
+		types::datatime_t currTime = types::dataclock::now();
+		const navtypes::Vectord<1> currPos{getMotorPos().getData()};
+		navtypes::Vectord<1> posCommand = velController->getCommand(currTime, currPos);
+		// TODO: is there only one position value in the matrix?
```

Comment:
Yeah, since all the dimensions are one, there's only one. So this todo can be removed